tests/integration_tests/performance/test_net_tput.py

```diff
@@ -0,0 +1,278 @@
+    }
+
+
+def run_iperf(microvm, host_ip, results, iperfs, protocol):
+    """Run tests using given protocol."""
+    run_statistics = {}
+
+    for _ in range(RUNS):
+        for bw in BANDWIDTHS[protocol]:
+            for mode in MODES.items():
+                for window_size in range(2):
+                    for packet_size in PACKET_SIZE_KB:
+                        cmd = '{} -c {} -t{} -f MBytes -J -b {}M {} {} -A {} \
+                                -P {} {} {} {}'.format(
+                            IPERF,
+                            host_ip,
+                            ITERATION_DURATION,
+                            bw,
+                            mode[1],
+                            PROTOCOL[protocol],
+                            PINNING_CPU,
+                            iperfs,
+                            '-w {}'.format(WINDOW_SIZE_MB) if protocol == 'tcp' and window_size == 0  else '',
+                            '-O {}'.format(OMIT_TIME) if protocol == 'tcp' else '',
+                            '-l {}'.format(packet_size) if protocol == 'tcp' else '',
+                        )
+
+                        # Don't do multiple runs on udp with the same parameter.
+                        if packet_size > PACKET_SIZE_KB[0] and protocol == 'udp':
+                            continue
+                        if window_size == 1 and protocol == 'udp':
+                            continue
+
+                        iperf_out = spawn_guest_iperf(microvm, cmd)
+                        if protocol == 'tcp':
+                            result = parse_iperf_tcp_result(iperf_out, mode[0],
+                                                            bw, window_size,
+                                                            packet_size, run_statistics)
+                        else:
+                            result = parse_iperf_udp_result(iperf_out, mode[0],
+                                                            bw, window_size, 
+                                                            run_statistics)
+                        results.append(result)
+                        print(result)
+
+    return run_statistics
+
+
+def get_cpus_in_numa(numa_node, vcpus):
+    """Return a list of #vcpus on given @numa_node."""
+
+    _p = run('lscpu | grep node{}'.format(numa_node), shell=True, 
+            universal_newlines=True, stdout=PIPE, stderr=PIPE)
+    assert _p.stdout is not ''
+
+    # Get the list of cpus in the numa node
+    numas = _p.stdout.split(':')[1].strip()
+    
+    cpu_lists = numas.split(',')
+    cpus = []
+    for cpu_list in cpu_lists:
+        print(cpu_list)
+        start, end = cpu_list.split('-')
+        start = int(start)
+        end = int(end)
+        if end - start + 1 < vcpus:
+            vcpus -= end - start + 1
+            cpus.extend(range(start, end + 1))        
+        else:
+            cpus.extend(range(start, start + vcpus))
+            vcpus = 0
+            break
```

Comment:
this logic seems quite complicated. Maybe it would be easier to use the info in  `cat /sys/devices/system/node/node*/cpumap` or `cat /sys/devices/system/node/node*/cpulist` .